### PR TITLE
Migration guide from PHP 8.0 to PHP 8.1

### PR DESCRIPTION
**Pull request by Girgias**
Currently missing sections:

 - [x] 1. Backward Incompatible Changes
    - [x] Core 
    - [x] MySQLi
    - [x] Standard
 - [x] 2. New Features
    - [x] Hash 
    - [x] MySQLi
 - [x] 4. Deprecated Functionality

From the [UPGRADING](https://github.com/php/php-src/blob/PHP-8.1/UPGRADING) document in the PHP-8.1 branch.

### CODE REVIEW

File: appendices/migration81/deprecated.xml [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    Static methods and properties should only be accessed on a class using the trait.
+   </para>
+  </sect3>
+
+  <sect3 xml:id="migration81.deprecated.core.magic-sleep">
+   <title>Returning a non-&array; from <function>__sleep</function></title>
+
+   <para>
+    Returning a value which is not an <type>array</type> from
+    <function>__sleep</function> now generates a diagnostic.
```
Comment (kamil-tekiela) [inline appendices/migration81/deprecated.xml:80]: I can't find this change. The warning/notice has been present in previous versions. What changed exactly in PHP 8.1?
@krakjoe The entry was added by you it seems.

Comment (krakjoe) [reply appendices/migration81/deprecated.xml:80]: No recollection of what this is about, are you sure it was me ?

Comment (kamil-tekiela) [reply appendices/migration81/deprecated.xml:80]: https://github.com/php/php-src/commit/ed88daecf49a281e2ca3f40cbf7be19e22a57102

File: appendices/migration81/deprecated.xml [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+ <title>Deprecated Features</title>
+
+ <sect2 xml:id="migration81.deprecated.core">
+  <title>PHP Core</title>
+
+  <sect3 xml:id="migration81.deprecated.core.serialize-interface">
+   <title>
+    Implementing <interfacename>Serializable</interfacename> without
+    <function>__serialize</function> and <function>__unserialize</function>
+   </title>
```
Comment (cmb69) [inline appendices/migration81/deprecated.xml:12]: Can this title be shortened?

Comment (Girgias) [reply appendices/migration81/deprecated.xml:12]: Well other than dropping `` and <function>__unserialize</function>`` not sure, but I welcome any other title.

File: appendices/migration81/other-changes.xml [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    <methodname>ReflectionProperty::setAccessible</methodname> and
+    <methodname>ReflectionMethod::setAccessible</methodname>
+    no longer have an effect.
+    Properties and methods are now always considered accessible via Reflection.
+   </para>
+   <para>
+    <simplelist>
+     <member><constant>ReflectionClassConstant::IS_PUBLIC</constant></member>
+     <member><constant>ReflectionClassConstant::IS_PROTECTED</constant></member>
+     <member><constant>ReflectionClassConstant::IS_PRIVATE</constant></member>
```
Comment (mumumu) [inline appendices/migration81/other-changes.xml:94]: These constants seems to be unrelated to migration81 changes.
Maybe, copy-and-pasted from migration80 changes?

```
     <member><constant>ReflectionClassConstant::IS_PUBLIC</constant></member>
     <member><constant>ReflectionClassConstant::IS_PROTECTED</constant></member>
     <member><constant>ReflectionClassConstant::IS_PRIVATE</constant></member>
```

Comment (Girgias) [reply appendices/migration81/other-changes.xml:94]: Oh, indeed that's a left over.

File: appendices/migration81/new-features.xml [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+
+ <sect2 xml:id="migration81.new-features.curl">
+  <title>CURL</title>
+  
+  <para>
+   Added the <constant>CURLOPT_DOH_URL</constant> option.
+  </para>
+
+  <para>
+   Added options for blob certificate when libcurl >= 7.71.0:
```
Comment (mumumu) [inline appendices/migration81/new-features.xml:162]: `libcurl &gt;= 7.71.0` is better?

Comment (cmb69) [reply appendices/migration81/new-features.xml]: Could also use `≥`.
